### PR TITLE
JBEHAVE-1492 Make function parameter converter generic

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java [hunk 1]
```diff
@@ -617,23 +622,37 @@ private static boolean isAssignable(Type from, Type to) {
...
 
         @Override
-        public T convertValue(String value, Type type) {
+        public T convertValue(S value, Type type) {
             return converterFunction.apply(value);
         }
+
+    }
+
+    public static class FromStringFunctionalParameterConverter<T> extends FunctionalParameterConverter<String, T> {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java]: why is it needed?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java (hunk 1 above)
Comment (uarlouski) [reply jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java]: discussed privately

File: jbehave-weld/src/test/java/org/jbehave/core/configuration/weld/ConfigurationProducer.java [hunk 2]
```diff
@@ -63,19 +62,5 @@ public MyDateConverter() {
             super(new SimpleDateFormat("yyyy-MM-dd"));
         }
     }
-    
-    public static class CustomObject {
```
Comment (valfirst) [inline jbehave-weld/src/test/java/org/jbehave/core/configuration/weld/ConfigurationProducer.java]: what is the reason to remove this class?

Comment (uarlouski) [reply jbehave-weld/src/test/java/org/jbehave/core/configuration/weld/ConfigurationProducer.java]: just checking :)

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/29e12af3a339b852a47c31403737b18358a2efad

**Pull request closed without merging**